gammapy/estimators/map/ts.py

```diff
@@ -424,20 +428,36 @@ def estimate_fit_input_maps(self, dataset):
             dataset=dataset, kernel=kernel, exposure=exposure
         )
 
+        mask_safe = dataset.mask_safe if dataset.mask_safe else 1.0
+        counts = dataset.counts * mask_safe
+        background = dataset.npred() * mask_safe
+        exposure *= mask_safe
+
         energy_axis = counts.geom.axes["energy"]
 
         flux_ref = self.model.spectral_model.integral(
             energy_axis.edges[0], energy_axis.edges[-1]
         )
 
         exposure_npred = (exposure * flux_ref * mask.data).to_unit("")
-
         norm = (flux / flux_ref).to_unit("")
+
+        if self.sum_over_energy_groups:
+            if dataset.mask_safe is None:
+                mask_safe = Map.from_geom(counts.geom, data=True, dtype=bool)
```

Comment:
yes here we need the energy axis for the `mask_safe` because it will be used as weight to compute the 2d kernel, this is done later in `from_arrays` because the mask_safe region to multiply with the kernel depends on the position:
```
            mask_safe = _extract_array(mask_safe.data, kernel.shape, position)
            kernel = (kernel * mask_safe).sum(axis=0, keepdims=True)
            with np.errstate(invalid="ignore", divide="ignore"):
                kernel /= mask_safe.sum(axis=0, keepdims=True)
                kernel[~np.isfinite(kernel)] = 0
```